src/components/common/BtnTask/BtnTask.tsx

```diff
@@ -0,0 +1,214 @@
+import Icons from '@/assets/svg/index';
+
+interface BtnTaskProps {
+	btnType: number;
+	status?: string;
+	isDescription?: boolean;
+}
+
+interface BorderColorProps {
+	isHovered: boolean;
+	isClicked: boolean;
+	iconHovered: boolean;
+	theme: Theme;
+	btnType: number;
+}
+
+function BtnTask(props: BtnTaskProps) {
+	const { btnType, status = 'Todo', isDescription = false } = props;
+	const [isClicked, setIsClicked] = useState(false);
+	const [isHovered, setIsHovered] = useState(false);
+	const [iconHovered, setIconHovered] = useState(false);
+	const [iconClicked, setIconClicked] = useState(false);
+
+	const handleMouseEnter = () => {
+		setIsHovered(true);
+	};
+
+	const handleMouseLeave = () => {
+		setIsHovered(false);
+	};
+
+	const handleClick = () => {
+		setIsClicked((prev) => !prev);
+	};
+
+	const stopPropagation = (e: React.MouseEvent) => {
+		e.stopPropagation();
+	};
+
+	const handleIconMouseEnter = () => {
+		setIconHovered(true);
+	};
+
+	const handleIconMouseLeave = () => {
+		setIconHovered(false);
+	};
+
+	const handleIconClick = (e: React.MouseEvent) => {
+		if (iconClicked) {
+			setIconClicked(false);
+		} else {
+			stopPropagation(e);
+			setIconClicked(true);
+		}
+	};
+
+	const renderStatusButton = () => {
+		if (btnType === 3) {
+			return <StagingIconHoverIndicator />;
+		}
+
+		if (!iconHovered && !iconClicked) {
+			return <IconHoverIndicator />;
+		}
+
+		if (btnType === 1) {
+			return <StatusStagingBtn />;
+		}
+
+		if (btnType === 2) {
+			if (status === 'Done') {
+				return <StatusDoneBtn />;
+			}
+			if (status === 'InProgress') {
+				return <StatusInProgressBtn />;
+			}
+			return <StatusTodoBtn />;
+		}
+
+		return null;
+	};
+
+	return (
+		<BtnTaskLayout
+			isClicked={isClicked}
+			isHovered={isHovered}
+			iconHovered={iconHovered}
+			btnType={btnType}
+			onClick={handleClick}
+		>
+			<BtnTaskContainer onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+				<BtnTaskTextWrapper isDescription={isDescription}>
+					{isDescription && <IconFile />}
+					넛쉘 UT 진행하기
+				</BtnTaskTextWrapper>
+				{btnType === 3 ? (
+					<BtnDate date="2024.07.11" time="22:22" size="small" isDelayed />
+				) : (
+					<BtnDate date="2024.07.11" time="22:22" size="small" />
+				)}
+			</BtnTaskContainer>
+			<IconHoverContainer
+				onClick={handleIconClick}
+				onMouseEnter={handleIconMouseEnter}
+				onMouseLeave={handleIconMouseLeave}
+			>
+				{renderStatusButton()}
+			</IconHoverContainer>
+		</BtnTaskLayout>
+	);
+}
+
+export default BtnTask;
+
+const getBorderColor = ({ isHovered, isClicked, iconHovered, theme, btnType }: BorderColorProps) => {
+	let borderColor;
+
+	if (btnType === 3) {
+		borderColor = theme.palette.Orange.Orange8;
+	} else if (iconHovered || isHovered) {
+		borderColor = theme.palette.Blue.Blue1;
+	} else if (isClicked) {
+		borderColor = theme.palette.Primary;
+	} else {
+		borderColor = theme.palette.Grey.Grey1;
+	}
+
+	return css`
+		border-color: ${borderColor};
+	`;
+};
```

Comment:
```suggestion
const getBorderColor = ({ isHovered, isClicked, iconHovered, theme, btnType }: BorderColorProps) => {
	const defaultColor = theme.palette.Grey.Grey1;
	const hoverColor = theme.palette.Blue.Blue1;
	const clickColor = theme.palette.Primary;
	const orangeColor = theme.palette.Orange.Orange8;
	let borderColor = defaultColor;

	if (btnType === 3) {
		borderColor = orangeColor;
	} else if (iconHovered || isHovered) {
		borderColor = hoverColor;
	} else if (isClicked) {
		borderColor = clickColor;
	}

	return css`
		border-color: ${borderColor};
	`;
};
```

이런식으로 조금 더 가독성좋게 써볼 수도 있을 것 같아요~!
저 테마 변수로 설정하는 건 이 컴포넌트 범위로 빼서 계속 사용할 수도 있을 것 같네요

Reply:
오,, 위에 띰만 빼줬는데 훨씬 가독성이 좋아졌네요.,,